libkbfs/keybase_daemon_rpc.go

```diff
@@ -255,47 +277,28 @@ func (k *KeybaseDaemonRPC) AddProtocols(protocols []rpc.Protocol) {
 	} else {
 		k.protocols = protocols
 	}
+
+	// If we are already connected, register these protocols.
+	k.lock.RLock()
```

Comment:
Is `k.lock` supposed to be protecting `k.protocols` as well?  If so, the above code modifies it outside the lock.  (Of course that problem was pre-existing, but since you added `k.lock` above `k.protocols` in the struct, it's now a little more confusing.

Reply:
good catch!